Model Registry migration (#1430)
wandb/docs: content/en/guides/core/registry/model_registry_eol.md

## Preparing for the migration

No action is required from users before the migration. W&B will handle the transition, ensuring that workflows and references are preserved.

## During the migration

W&B will initiate the migration process. The migration will occur during a time window that minimizes disruption to W&B services. There will be a brief pause (estimated around 2-3 hours) on write operations for a specific team's assets. This means that while your team's legacy Model Registry data is being copied to the new W&B Registry, you won't be able to save new versions or make changes in either that specific legacy Model Registry or its new Registry counterpart. Once the copy is complete for your team, write operations will resume normally in the new W&B Registry. The legacy Model Registry will then become read-only. 

The following will happen during the migration:

* W&B will migrate registered models (now called *collections*) and associated artifact versions [from the legacy Model Registry to the W&B Registry]({{< relref "#artifact-paths" >}}). 

> * API calls that reference the legacy Model Registry will be [redirected to the new W&B Registry]({{< relref "#legacy-model-registry-apis" >}}). This means that any existing references to model artifacts in your code or workflows will continue to function without requiring updates.

Reviewer: **suggestion:** I'd suggest adding a caveat at the end of this point recommending the users to update their scripts with paths from the new registry

> * Team level visibility of the legacy Model Registry will be [transitioned to organization level visibility]({{< relref "#team-visibility-to-organization-visibility" >}}) in the W&B Registry.

Reviewer: Actually, the new organization-level registry will have `restricted` visibility, where the team whose model registry is migrated will be added as a member and the team and model registry admins will be added to the new registry as admins.

Reviewer: But they can change those settings after the migration is complete

Reviewer: Unless you meant to say that we're moving a team's model registry to the organization-level..?

* The legacy Model Registry will transition to a [read-only state]({{< relref "#legacy-model-registry-read-only-state" >}}) once the migration begins and will remain accessible for reference.

## After the migration

Post-migration, collections, artifact versions, and associated attributes will be fully accessible within the new W&B Registry. The focus is on ensuring that current workflows remain intact, with ongoing support available to help navigate any changes.

### Artifact paths

Legacy model registry path will be redirected to the new W&B Registry path. This means that any existing references to model artifacts in your code or workflows will continue to function without requiring updates.

For example, if you have a model artifact path in your code like:

```python
model_reg_path = "team-name/model-registry/collection-name:version"
```

> W&B will automatically redirect to its new W&B Registry path:

Reviewer: ```suggestion
W&B will automatically fetch the version from the new W&B Registry path:
```

```python

> registry_path = "org-name/wandb-registry-team-name/collection-name:version"

Reviewer: ```suggestion
registry_path = "org-name/wandb-registry-<team-name>-mr-migrated/collection-name:version"
```

```

### Legacy Model Registry APIs

The following API calls will be supporting automatic redirects to the new W&B Registry:

* `wandb.Api().artifact()`
* `wandb.Api().use_artifact()`
* `wandb.run.link_artifact()`
* `wandb.Artifact.link()`

{{% alert title="Python SDK version"%}}
Based on your W&B Python SDK version, the redirection behavior will vary:

* Users with Python SDK versions v0.22 and greater will receive a non-breaking warning in your SDK logs about this redirection, encouraging users to update paths in new code over time.
* Users with Python SDK versions below v0.22 will not receive a warning, but the redirection will still occur.

{{% /alert %}}

> <!-- Add note about degraded experience? If yes, get more concrete details from the engineering team. -->

Reviewer: Will defer to @trane293 / @tristan-spaulding2 on whether we should include details. Happy to provide more info once we have that confirmation!

### Team visibility to organization visibility

> After the migration, your model registry will have organization-level visibility. You can restrict who has access to a registry by [assigning roles]({{< relref "./configure_registry.md" >}}). This helps ensure that only specific members have access to specific registries.

Reviewer: Similar comment as above

W&B will attempt to preserve existing permission boundaries of your current team-level registered models in the legacy W&B Model Registry. W&B suggests that admins to review access to the new migrated organization-level registries to ensure they align with your expectations.

### Legacy Model Registry read-only state

> The legacy Model Registry will transition to a read-only state once the migration begins. This means that while you can still view and reference your existing registered models, you will not be able to make any changes or updates to them in the legacy Model Registry. All new model registrations and updates will need to be done in the new W&B Registry.

Reviewer: ```suggestion
The legacy Model Registry will transition to a read-only state once the migration begins. This means that while you can still view and reference your existing registered models, you will not be able to make any changes or updates to them in the legacy Model Registry. All new collection creation and updates will need to be done in the new W&B Registry.
```

## FAQs

### How much notice will we receive before our organization is scheduled for migration?

> W&B is committed to providing sufficient notice. Expect to receive direct communication and see an advisory banner in the W&B UI approximately 4-6 weeks before your organization's scheduled migration window. This will give your team time to prepare and ask any questions. 

Reviewer: Will defer to @trane293 on this, but looks like we're aiming for a 2 week notice now

### How long will copying data from the legacy Model Registry to the W&B Registry take?
This depends on the volume of data in your legacy Model Registry. During this brief copy period for your specific team, write access to that registry (both the old Model Registry and the new W&B Registry) will be temporarily paused to ensure data integrity. W&B anticipates this specific write-pause to be around an hour or so based on current testing. W&B will provide more specific estimates during targeted communications. Other W&B services will remain fully operational.

### What if we use features like Protected Aliases or Team Service Accounts heavily?
The migration is designed to handle these. Protected Aliases will have an equivalent in the W&B Registry, and your existing ones will be mapped. Team Service Account permissions will be preserved within the context of the migrated registries.

### How long will we be able to access our legacy Model Registry (MR) after our organization is migrated?

> After your organization's data is successfully copied to the new W&B Registry (GR), the legacy MR will immediately become read-only. You will be able to view your data in the legacy MR in this read-only state for approximately 4 weeks. After this period, the UI views for the legacy MR will be hidden.

Reviewer: @trane293 We're already hiding the UI for the legacy model registry post migration. Did we want to mention that users can view their data from the `model-registry` project view? Or should we avoid that?
